deepchem/models/torch_models/layers.py

```diff
@@ -2782,3 +2782,135 @@ def forward(
             else:
                 out_tensor += w * in_tensor
         return out_tensor
+
+
+class EdgeNetwork(nn.Module):
+    """ Submodule for Message Passing
+
+    Examples
+    --------
+    >>> pair_features = torch.rand((4, 2), dtype=torch.float32)
+    >>> atom_features = torch.rand((5, 2), dtype=torch.float32)
+    >>> atom_to_pair = []
+    >>> n_atoms = 2
+    >>> start = 0
+    >>> C0, C1 = np.meshgrid(np.arange(n_atoms), np.arange(n_atoms))
+    >>> atom_to_pair.append(np.transpose(np.array([C1.flatten() + start, C0.flatten() + start])))
+    >>> atom_to_pair = torch.Tensor(atom_to_pair)
+    >>> atom_to_pair = torch.squeeze(atom_to_pair.to(torch.int64), dim=0)
+    >>> inputs = [pair_features, atom_features, atom_to_pair]
+    >>> n_pair_features = 2
+    >>> n_hidden = 2
+    >>> init = 'xavier_uniform_'
+    >>> layer = EdgeNetwork(n_pair_features, n_hidden, init)
+    >>> result = layer(inputs)
+    >>> result.shape[1]
+    2
+    """
+
+    def __init__(self,
+                 n_pair_features: int = 8,
+                 n_hidden: int = 100,
+                 init: str = 'xavier_uniform_',
+                 **kwargs):
+        """initalise a EdgeNetwork Layer
+        Parameters
+        ----------
+        n_pair_features: int, optional
+            The length of the pair features vector.
+        n_hidden: int, optional
+            number of hidden units in the passing phase
+        init: str, optional
+            Initialization function to be used in the message passing layer.
+        """
+
+        super(EdgeNetwork, self).__init__(**kwargs)
+        self.n_pair_features: int = n_pair_features
+        self.n_hidden: int = n_hidden
+        self.init: str = init
+
+        init_func: Callable = getattr(initializers, self.init)
+        self.W: torch.Tensor = init_func(
+            torch.empty([self.n_pair_features, self.n_hidden * self.n_hidden]))
+        self.b: torch.Tensor = torch.zeros((self.n_hidden * self.n_hidden,))
+        self.built: bool = True
+
+    def __repr__(self) -> str:
+        return (
+            f'{self.__class__.__name__}(n_pair_features:{self.n_pair_features},n_hidden:{self.n_hidden},init:{self.init})'
+        )
+
+    def forward(self, inputs: List[torch.Tensor]) -> torch.Tensor:
+        """
+        Parameters
+        ----------
+        inputs: List[torch.Tensor]
+            The length of atom_to_pair should be same as n_pair_features.
+
+        Returns
+        -------
+        out_tensor: torch.Tensor
+        """
+        pair_features: torch.Tensor
+        atom_features: torch.Tensor
+        atom_to_pair: torch.Tensor
+        pair_features, atom_features, atom_to_pair = inputs
+
+        def unsorted_segment_sum(data, segment_ids, num_segments):
+            """
+            Computes the sum along segments of a tensor. Analogous to tf.unsorted_segment_sum.
+            :param data: A tensor whose segments are to be summed.
```

Comment:
Make sure to use numpy docstyle (https://numpydoc.readthedocs.io/en/latest/format.html)